src/ParallelAlgorithms/Events/ObserveFields.hpp

```diff
@@ -223,6 +224,17 @@ class ObserveFields<VolumeDim, ObservationValueTag, tmpl::list<Tensors...>,
             NonSolutionTensors::type::size()...},
         0_st));
 
+    // interpolated coordinates can be calculated exactly with element_map
+    auto mapped_coordinates =
+        interpolation_mesh_.has_value()
+            ? element_map(logical_coordinates(*interpolation_mesh_))
```

Comment:
This gives the wrong answer in evolutions where we have a moving mesh because we have 2 maps. I don't see why this is necessary, since the error in the coordinates will be at truncation error and so the difference isn't meaningful as far as I can tell.

Reply:
We found the error to be pretty significant. 
Presumably this is because the interpolation is essentially just evaluating a Lagrange polynomial with order equal to the number of source collocation points. While this would be exact on the logical coordinates, once it is not on the inertial coordinates because the polynomial cannot exactly capture the element map. Its probably only exact (up to truncation) if the map from logical to inertial coordinates is of lower order than the Lagrange polynomial. However this map can of course not even be represented exactly by a polynomial in many cases.

I am not yet really familiar with all the different maps going on in spectre but I can take a look to see if there is an easy solution.

Reply:
It doesn't matter how "significant" you think it is, the interpolated coordinates are just as correct as the mapped ones. If the relative truncation error is 0.03, then if the analytically mapped coordinate value is 4.31, and the interpolated value is 4.32, then they are the same number, you can't distinguish between them. The elliptic solver or evolution scheme picks up error from not being able to represent the map, and that means this level of inaccuracy is also present in the other variables printed out. I see no reason to jump through hoops to give different approximation to the coordinates than to any other variable. I personally would take the stance that the interpolated value is actually the correct value since it's evaluated by the approximation to the analytic map (say the polynomial map), and it is this polynomial map that the numerical scheme uses to solve for the data.

Reply:
Interpolating the coords means that if we interpolate one run at different resolutions) to a common output grid (e.g. to take differences), then the output data will have every so slightly different coordinates.  This may be a problem for visualizations- I don't know whether paraview allows to subtract to datasets for which the coords do not agree.

Interpolating the coordinates may also bring coordinates outside expected bounds.  E.g. a point on the inner boundary may move into the interior of the excised region.  A point may be shifted into a neighboring element.  The coordinates of the identical boundary-point shared by two elements will be  different after interpolation.

Finally, I agree that the simulation will have truncation error.  But as it happens, we do know precisely the inertial location of the points to which we interpolate: They are the mapped logical coordinates.  


**However**, having said all this in favour of mapped coordinates, I actually **do not** like this PR:  The interpolator does two things:
  1. It takes a list of user-specified points and a list of variables, and it interpolates the variables to the user-specified points.   _If inertial coordinates are included in the list of variables, then they should be interpolated as any other variable_
  2. The interpolator may (or may not) associate "coordinates" with each interpolated point (i.e. the "Coords" of Paraview).  The choice of what to use as 'coordinates' in the output is independent of whether coordinates may have been interpolated and which ones that may have been.  

Currently, this code seems to conflate these two things.  The code simply assumes that inertial_coordinates are the ones to use as coordinates.  

I actually don't have a good suggestion of how to proceed.

Reply:
Hi Harald, thanks for your input. Some comments from me:

1. The inertial coordinates are always written to the output file regardless whether the user specifies it or not. This PR does not change that but merely writes the exact mapping of the logical coordinates instead.
2. The GenerateXdmf script  which transforms the h5 file to xdmf as read by paraview expects there to be coordinates written to the h5 file, by default they are the inertial coordinates. I am not 100% certain  but believe that paraview will then use exactly these coordinates to plot the tensor components.

Reply:
Some quick thoughts:

- It sounds like part of the struggle is having ParaView compute the error between resolutions. Maybe there's a way to tell ParaView what tolerance it should use to associate points? Maybe the error can be computed before hand and written to a file (ApplyObservers-style)? we know that the grid points are aligned (roughly anyway) and so we can just associate the values at the grid points
- While I agree that the interpolation may cause points to be "outside" the element, I'm not sure what the correct way of defining the "boundary of an element" is. The numerical scheme takes numerical derivatives, and so in that sense the map is the polynomial projection onto the basis, but this projection will be ever so slightly different between neighboring elements. The "correct" way around this would be isoparametric maps at the Block level (well out of the scope of this PR), and then using uniform `p` across all elements (correct for evolutions, I think, but probably not elliptic problems since DG+AMR seems to actually be able to handle discontinuities in elliptic problems :D )
  Also, because we effectively use a projection of the map onto the local basis, I'm not sure I agree that the precise inertial points are the mapped logical coordinates. These are reasonable inertial points, but they are not what the numerical scheme uses, except at the points on the `Mesh` (i.e. the collocation points). If we use Gauss points, I think my argument becomes clearer: we interpolate the fields to the logical boundaries, accepting that this will introduce some small error, compared to having infinite basis functions. It's the exact same error we pick up here when we interpolate the coordinates
- re point 2 from @nikwit: yes, this is what ParaView does (or at least what we are telling it to do in the XDMF files). One could write the Grid frame points as well
- My biggest concern with the PR in its current form is that it doesn't work for the evolution code. I'll elaborate in a separate list below

The difference between elliptic and evolution:
- In the evolution code we can move the mesh. There is the `Inertial` frame which maps the coordinates frame the logical frame under some time-dependent mapping.  However, not all these maps are time-dependent, some are time-independent. This is where the grid frame comes in. The grid frame is a mapping from the logical coordinates where all the maps are time-independent (and we can cache the output). This means the evolution code actually has 2 maps 1) `ElementMap` going from logical->Grid and 2) `CoordinateMap` going from Grid->Inertial.
- We should write the "correct" inertial coordinates in both the elliptic and evolution code. As it stands it looks like this PR only will work for the elliptic code, but not the evolution code. This likely isn't caught because `PlaneWave1DObserveExample.yaml` doesn't have a time-dependent map. If you added a uniform translation, I would expect this PR to not pass tests.
- A related issue: what happens if I want to observe some other coordinates? E.g. `Distorted`. Should we interpolate those or calculate them from the coordinate maps? (I'm not saying we can't do either of these, I'm just saying a consistent choice should be made :) )

I honestly don't know how to support observing different coordinates without interpolation since in general you won't have a map going from the logical coordinates, but some other intermediate frame. The `FunctionsOfTime` (how the time-dependent maps determine their time-dependence) also would need to be passed to `ObserveFields`, along with the actual current time value.

Overall, I think this change is a reasonable thing to want, but it raises a lot of subtle difficulties that need to be correct before this could be merged, the biggest is support for moving meshes.